hail/python/test/hail/matrixtable/test_matrix_table_from_parts.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+import hail as hl
+
+class Inits():
+    globals={'hello': 'world'}
+    rows={'foo': ['a', 'b']}
+    cols={'bar': ['c', 'd']}
+    entries={'baz': [[1, 2], [3, 4]]}
+
+def unless(test: bool, kvs):
+    return {} if test else kvs
+
+def assert_matches_globals(mt: 'hl.MatrixTable', no_props=False):
+    assert hl.eval(mt.globals) == hl.Struct(**unless(no_props, Inits.globals))
+
+def assert_matches_rows(mt: 'hl.MatrixTable', no_props=False):
+    assert mt.rows().order_by('row_idx').collect() == \
+        [ hl.Struct(row_idx=0, **unless(no_props, {'foo':'a'}))
+        , hl.Struct(row_idx=1, **unless(no_props, {'foo':'b'}))
+        ]
+
+def assert_matches_cols(mt: 'hl.MatrixTable', no_props=False):
+    assert mt.cols().order_by('col_idx').collect() == \
+        [ hl.Struct(col_idx=0, **unless(no_props, {'bar': 'c'}))
+        , hl.Struct(col_idx=1, **unless(no_props, {'bar': 'd'}))
+        ]
+
+def assert_matches_entries( mt: 'hl.MatrixTable'
+                          , no_row_props=False
+                          , no_col_props=False
+                          , no_entry_props=False
+                          ):
+    assert mt.entries().order_by('row_idx', 'col_idx').collect() == \
```

Comment:
Since you're really trying to check that `from_parts` has the right entries (in particular, you're not testing the `entries` method itself) and since you separately check the row and column annotations, I think you're totally clear to do:

```
actual = mt.select_rows().select_cols().entries()
assert actual == [hl.Struct(row_idx=0, col_idx=0, ...), ...]
```

The "select" methods [1] keep only the relevant key and those fields provided as arguments.

[1] `Table.select_globals`, `Table.select`, `MatrixTable.select_rows`, `MatrixTable.select_cols`, `MatrixTable.select_globals`, `MatrixTable.select_entries`

Reply:
Ah good to know, nice to know this exists!